### PR TITLE
SCons: Use colored output if available, change "colored"->"verbose"

### PR DESCRIPTION
**Pull request by zaps166**
Detect whether GCC supports colored output and use it.

### CODE REVIEW

**Comment by zaps166** [comment]
The alternative is to use:
```python
if 'TERM' in os.environ:
	env_base['ENV']['TERM'] = os.environ['TERM']
```
Works for both: GCC and Clang. Also it could be used without `colored` option (use default compiler settings and it will show colored output with full output without `Compiling ==> ...`), what do you think?

**Comment by akien-mga** [comment]
I'm not for making it a default as it hides a lot of information, like the used CXXFLAGS and CPPPATH. But it looks nice when activated :)

**Comment by zaps166** [comment]
> I'm not for making it a default as it hides a lot of information

Yes, but I thought about colored output with all information as default. And `colored` option name could be changed in this case - it will only show less output :)

**Comment by akien-mga** [comment]
Oh I hadn't checked that it was `colored` that hid so much stuff (was indeed surprising to see a gcc flag reduce verbose so much :D). I guess splitting the two would be good yes, and having colored + verbose by default would be fine IMO.

**Comment by zaps166** [comment]
Ok, so which name will be good for `colored` option which will only reduce output :) ?

**Comment by akien-mga** [comment]
`verbose=no` :)

**Comment by zaps166** [comment]
Tested on Linux and OS X. On Windows it doesn't make any problems (output on VS2015 is not colored; cannot build the newest Godot using MinGW on Windows - why?) :)

File: SConstruct [hunk 1]
```diff
@@ -65,6 +65,8 @@ elif (os.name=="nt"):
 		custom_tools=['mingw']
 
 env_base=Environment(tools=custom_tools);
+if 'TERM' in os.environ:
+	env_base['ENV']['TERM'] = os.environ['TERM']
```
Comment (akien-mga) [inline SConstruct]: What does this part do?

Comment (akien-mga) [inline SConstruct]: I see you mentioned earlier that it was sufficient to get colored output, but how does it work? My `$TERM` is `xterm`, but that's not the terminal I'm using. Is that sufficient to trigger a coloured output in a different terminal?

Comment (akien-mga) [inline SConstruct]: Ok, tested, it seems to work in konsole/yakuake.

**Pull request merged**